docs: Update status and faq, including entry on app store (#121)
shorebirdtech/docs: docs/faq.md

https://stackoverflow.com/questions/53291007/can-an-android-application-identify-the-test-track-within-google-play
https://stackoverflow.com/questions/26081543/how-to-tell-at-runtime-whether-an-ios-app-is-running-through-a-testflight-beta-i

> If you'd like to segment availability of Shorebird patches, we have 4 options:

Reviewer: ```suggestion
If you'd like to segment availability of Shorebird patches, there are 4 potential options:
```
